pkg/controllers/management/clusterdeploy/clusterdeploy.go

```diff
@@ -389,6 +387,8 @@ func (cd *clusterDeploy) deployAgent(cluster *apimgmtv3.Cluster) error {
 
 	cluster.Status.AppliedAgentEnvVars = append(settings.DefaultAgentSettingsAsEnvVars(), cluster.Spec.AgentEnvVars...)
 
+	cluster.Status.AppliedClusterAgentDeploymentCustomization = cluster.Spec.ClusterAgentDeploymentCustomization
```

Comment:
@jiaqiluo , in line with your other comment here, do we want to do DeepCopy here or are fine with simple assignment?